### DISCUSSION TITLE
More control over server island execution

### DISCUSSION THREAD

**Discussion by psmyrdek** [withastro/roadmap#1075] [Proposal]
# Summary

Introduce more flexibility around server island loading trigger (i.e. automatic vs depending on visibility, etc.).

# Background & Motivation

- When compared to multiple variants of client directives, server islands loading strategy is too rigid. For content-intense pages, there may be some secondary or tertiary elements rendered as server islands that should wait for their turn to be executed.

# Goals

- Addressing limited loading strategy when compared to client islands.

# Example

Just a few random ideas to kickstart the conversation:

```html

<!-- Default -->
<EpisodesGridAstro server:defer />

<!-- Similar to "lazy" img and iframe -->
<EpisodesGridAstro server:defer="lazy" />

<!-- More explicit  -->
<EpisodesGridAstro server:defer="viewport" />

<!-- Defer until another element is ready?  -->
<EpisodesGridAstro server:defer=".cart" />
```

**Comment by algernhon** [comment]
Something like the prefetch machanism would also be awesome : hover, tap ... That would gives Astro some Resumability vibes.
